### PR TITLE
Adding Rule book for Disk and NIC selection.

### PR DESCRIPTION
**Pull request by Ashughorla**
Added document for rules which will be used by user for Disk and NIC selection.
Related to [Proposal](https://github.com/metal3-io/metal3-docs/pull/152)

Kindly review.
@dhellmann @digambar15 @maelk @zaneb @russellb

### CODE REVIEW

**Review by jtwill98 — COMMENTED** [review]
Please add broadcom.

**Comment by Ashughorla** [comment]
> Please add broadcom.

Addressed

File: docs/rule_book.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+For `disk` selection user can use combination of `hctl` and
+`rotational` parameters.
+
+For `NIC` user have to provide vendor ID.
+
+## `HCTL`
+
+  Here `hctl` represents:
+
+  1. SCSI adapter number `host`
```
Comment (digambar15) [inline docs/rule_book.md:12]: Fix numbering

Comment (digambar15) [reply docs/rule_book.md:12]: all lines represent 1

File: docs/rule_book.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+## Vendor Dell
+
+   We are taking example of Dell hardware here.
+
+### Disk
+
+  1. Individual HDD/SSD : with `hctl` as 0:0:N:0 and rotational flag as
+  True/False.
+  1. PERC RAID of HDDs/SSDs : with `hctl` as 0:N:0:0/0:N:N:0 and rotational
+  flag as True.
```
Comment (digambar15) [inline docs/rule_book.md:32]: Same here! Fix numbering

File: docs/rule_book.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  1. Dell BOSS Controller Individual SSDs : with `hctl` as N:0:0:0 and
+  rotational flag as False.
+  1. Dell BOSS Controller Virtual Disk (RAID) : with `hctl` as N:0:0:0 and
+  rotational flag as True.
+  1. NVMe : No `hctl` Pattern, rotational flag as False and model name
```
Comment (digambar15) [inline docs/rule_book.md:37]: Ditto!

File: docs/rule_book.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  contains NVMe keyword.
+
+### NIC
+
+  1. Intel NIC Vendor ID is 0x8086.
+  1. Mellanox NIC Vendor ID is 0x15b3.
+  1. Broadcom NIC Vendor ID is 0x14e4.
```
Comment (digambar15) [inline docs/rule_book.md:44]: Ditto!

**Review by digambar15 — CHANGES_REQUESTED** [review]
Numbering of section contents should be fixed.

File: docs/rule_book.md (hunk 1 above)
Comment (zaneb) [reply docs/rule_book.md:12]: This is the normal way to do it in markdown. It renders as an ordered list in HTML, so the output numbering is always correct: https://github.com/metal3-io/hardware-classification-controller/pull/69/files?short_path=e0c0f2a#diff-e0c0f2a9ad43f0f94a2de56bb81a793415c04a2169f769eb9b5222333bd35c54

Comment (Ashughorla) [reply docs/rule_book.md:12]: Yes, numbering will look like that in .md file.

**Review by digambar15 — APPROVED** [review]
Looks good

**Pull request merged**